tensorboard/components/tf_backend/router.ts

```diff
@@ -62,7 +62,13 @@ export function createRouter(pathPrefix = 'data', demoMode = false): Router {
   };
 };
 
-let _router: Router = createRouter();
+export function getDefaultRouter(): Router {
+  const sep = window.location.pathname.endsWith('/') ? '' : '/';
```

Comment:
It’s not great that this introduces a side-effect at module load time.
We now need to reason about the first time that this module is imported;
it becomes harder to reason about in isolation.

Before #1512, we issued requests to relative paths. Can we just go back
to doing that instead of switching to absolute paths and then trying to
relativize them? We wouldn’t need to worry about `window.location` at
all, right?

Reply:
Noted. I haven’t carefully looked at the ways in which we use `URL`;
I’ll do so after the holiday.